### ISSUE TITLE
Sourcing the shell integration script is too slow for what it does

### ISSUE THREAD

**Issue by smancill** [shyiko/jabba#324]
My shell startup time goes from 100 ms to 340 ms just by sourcing the jabba shell integration file.

In other words, this takes 240 ms:

```sh
source "$JABBA_HOME/jabba.sh"
```

In comparison, _jenv_ takes around 100 ms without rehash:

```sh
eval "$(jenv init --no-rehash -)"
```

And *sdk* takes around 30 ms (but it may take longer with more versions installed?):

```sh
source "$SDKMAN_DIR/bin/sdkman-init.sh"
```

And of the three, jabba is actually the simpler one. The combination of the _alias_ and _use_ commands [to set the default Java version](https://github.com/shyiko/jabba/blob/3e8346c24eccc05c225fa24616aac683db01415b/install.sh#L116-L118) does something like this:

```sh
if [ -f "$JABBA_HOME/default.alias" ]; then
    jdk="$JABBA_HOME/jdk/$(cat "$JABBA_HOME/default.alias")"
    [ -d "$jdk/Contents/Home" ] && jdk="$jdk/Contents/Home"
    export JAVA_HOME_BEFORE_JABBA="$JAVA_HOME"
    export JAVA_HOME="$jdk"
    export PATH="$jdk:$PATH"
fi
```

But doing it with the jabba binary makes it an order of magnitude slower.

Not a Go developer, but this may be probably caused by the `init()` function configuring TLS for every subcommand, even when not needed.

**Comment by flexgrip** [comment]
Agreed. I had to remove it. It was causing my zsh boot time to be 1 second slower.
